### PR TITLE
Fix issue 9410 by removing special-case hack to suppress unknown attribute warnings on aptos_stdlib

### PR DESCRIPTION
**Pull request by brmataptos**
### Description

[PR 9229](https://github.com/aptos-labs/aptos-core/pull/9229) introduced warnings on unknown attributes, but suppressed them for aptos_stdlib to avoid user consternation about problems in the Aptos codebase.  It also removed those attributes from the codebase, so now that it is released it becomes safe to Fix i[ssue 9410](https://github.com/aptos-labs/aptos-core/issues/9410) and remove the special-case for the aptos_stdlib.

### Test Plan
Existing test cases, new test outputs.

### CODE REVIEW

File: third_party/move/move-compiler/tests/move_check/parser/aptos_stdlib_attributes.exp [hunk 1]
```diff
@@ -6,6 +18,18 @@ error[E02001]: duplicate declaration, item, or annotation
   │       │   
   │       Attribute previously given here
 
+warning[W02016]: unknown attribute
+  ┌─ tests/move_check/parser/aptos_stdlib_attributes.move:7:7
+  │
+7 │     #[testonly]
+  │       ^^^^^^^^ Attribute name 'testonly' is unknown (use --skip-attribute-checks CLI option to ignore); known attributes are '{"bytecode_instruction", "deprecated", "expected_failure", "native_interface", "test", "test_only", "verify_only"}'.
```
Comment (vineethk) [inline third_party/move/move-compiler/tests/move_check/parser/aptos_stdlib_attributes.exp:25]: Not a comment for this PR, but this seems like a good candidate for the move linter (to identify the closest allowed attribute and suggest it as fix).

**Review by vineethk — APPROVED** [review]
lgtm.

File: third_party/move/move-compiler/tests/move_check/parser/aptos_stdlib_attributes.exp (hunk 1 above)
Comment (brmataptos) [reply third_party/move/move-compiler/tests/move_check/parser/aptos_stdlib_attributes.exp:25]: Well, there is a list displayed.  Good point, though.

**Comment by brmataptos** [comment]
Note this PR was defective (was on my main instead of a branch), so I've closed it and am reopening as https://github.com/aptos-labs/aptos-core/pull/10175.

**Pull request closed without merging**